### DISCUSSION TITLE
I've made a PCB revision for audio switching and Dallas temperature probe

### DISCUSSION THREAD

**Discussion by chunkysteveo** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#752] [Ideas]
![image](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/14663002/a3b7408f-06d9-4ce2-a462-82be407f525f)
![image](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/14663002/7c78fc4f-ca90-4ae7-b906-809c4882fb98)

Made a revision of the PCB_V2 that includes a slide switch for audio selection - CIO pass through or from ESP (with an optional jumper if you're not using the resistor and capacitor filter). 

And also updated the pads for D8 and D0 to be 3V-D-G and a pull up resistor for easy inclusion of a Dallas temp probe for the ambient temp sensor. 

We'll see how they come out! If they work well, happy to open up the revision as share.

Has anyone done an SMD version of the board, with just the bare ESP and programming pins, rather than using a Node MCU module and through holes? It would really shrink the board down, you could even add in the LLC as passives to the board, rather an a module?

EDIT: To expand on the PCB updates above:

Audio Switching - With 6 wire models you can connect the buzzer to the ESP to get custom audio sounds (chirps, rings, melody up/down etc) when pressing buttons, and can also make the pump play melodies etc. This is at your own taste, and the sound is quieter. To get the original pump BEEEP on button presses, you need to bypass the buzzer wire and leave it "connected" from CIO to DSP. You can de-solder these two wires from the PCB and wire them up, but this new addition allows you to select the audio with all 6 wires soldered up on both CIO and DSP cables. Giving you a solder-free way to test if you want traditional BEEPS or new ESP "Chirps"!

Dallas Temp: To add in the ambient temperature probe (DS18B20) to allow the ESP to work out local ambient temp and use that for it's calculations. This update adds in the 4.7K Ohm pull up resistor between data and 3V for the DS18B20 needed, and also re-factors the pins to be 3V-D-G so you can just solder in the legs of a bare DS18B20 if you wanted to.

(The PDF says don't use D0 or D8, but the Web GUI says D0-D8??)

**Comment by visualapproach** [comment]
Dude! I just uploaded more songs for you and you make a switch to turn it off?! LoL 😂 
I've seen another project with SMD but not sure there is one on this. You're welcome to share this with us! Thanks

**Reply by chunkysteveo** [reply]
> Dude! I just uploaded more songs for you and you make a switch to turn it off?! LoL 😂 I've seen another project with SMD but not sure there is one on this. You're welcome to share this with us! Thanks

hahahah - i'll keep the switch on the ESP side when I solder it up, and test out your new melodies..... Tell me Rick Roll is on there?!?!!

**Reply by visualapproach** [reply]
Haha nope! May be royalties involved 😇

**Reply by chunkysteveo** [reply]
Just to check - 

The PDF says don't use D0 or D8 for the dallas temp, but the Web GUI says enter D0 to D8 for the temp pin. Are D8 and D0 OK to use? I was going to use D8, which seems OK?

Label: D8
GPIO: GPIO15
Input: pulled to GND
Output: OK
Notes: SPI (CS)Boot fails if pulled HIGH

- will boot fail on D8?

**Reply by visualapproach** [reply]
I haven't checked myself. From what I recall D0 is working. Doesn't the Dallas unit need a pull-up? In that case boot will fail?

**Reply by chunkysteveo** [reply]
> I haven't checked myself. From what I recall D0 is working. Doesn't the Dallas unit need a pull-up? In that case boot will fail?

Yeah, looking at the pin spec and the Dallas - a pull up is needed, and the ESP spec states D8 pulled up on boot will not boot.... I will test when I get them, and remove the custom pinout for D8 if that's the case. D0 *should* be OK as it is high at boot.....

Label: D0
GPIO: GPIO16
Input: no interrupt
Output: no PWM or I2C support
Notes: HIGH at boot. Used to wake up from deep sleep
